h2/src/main/org/h2/mode/FunctionsPostgreSQL.java

```diff
@@ -235,6 +247,13 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             // Not implemented
             result = v1.convertTo(Value.VARCHAR);
             break;
+        case ARRAY_TO_STRING:
+            StringJoiner joiner = new StringJoiner(String.valueOf(v1.getObject()));
+            for (Value v : ((ValueArray) v0).getList()) {
```

Comment:
You need to convert the type to `Value.ARRAY` first with `v0.convertTo(TypeInfo.TYPE_ARRAY)` or something like it.

Reply:
`array_to_string` is removed from this PR